File: DSA/YashKrishanGupta_590014798/Day_34_Question_3.java
import java.util.Scanner;
public class Day_34_Question_3 {
    // Solution 3
    public static boolean isBinaryPalindrome(int n){
        int left = numberOfBit(n)-1;
        int right = 0;
        while(left>right){
            int left_bit = (n >> left) & 1;
            int right_bit = (n >> right) & 1;
            if(left_bit!=right_bit) return false;
            left--;
            right++;
        }
        return true;
    }

    public static int numberOfBit(int n){
        int length = 0;
        while(n>0){
            length++;
            n>>=1;
        }
        return length;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter Integer : ");
        int num = sc.nextInt();
        System.out.print(isBinaryPalindrome(num));
    }
}

File: DSA/SakshamKashyap_590014183/day34/day34_Question_3.py
def is_binary_palindrome(n):
    num_bits = n.bit_length()
    for i in range(num_bits // 2):
        left_bit = (n >> (num_bits - i - 1)) & 1
        right_bit = (n >> i) & 1
        if left_bit != right_bit:
            return "No"
    return "Yes"

print(is_binary_palindrome(9))  
print(is_binary_palindrome(10))  
